packages/stencil/src/components/datacom-radio/datacom-radio.tsx

```diff
@@ -0,0 +1,181 @@
+  tag: 'datacom-radio',
+  styleUrl: 'datacom-radio.css',
+  scoped: true,
+})
+export class DatacomRadio implements FormControl {
+  @Element() host: HTMLElement;
+  @Prop() label: string;
+  @Prop() variant: RadioVariant = 'radios';
+  @Prop() size: RadioSize = 'standard';
+  @Prop({ attribute: 'image-position' }) imagePosition: ImagePosition = 'left';
+  @Prop() type = 'radio';
+  @Prop({ mutable: true }) checked = false;
+  @Prop() disabled = false;
+  @Prop() required = false;
+  @Prop() name: string;
+  @Prop() src: string;
+  @Prop() icon: string;
+
+  @Prop() value: string;
+  @Prop() autofocus = false;
+  @Prop() autocomplete?: boolean;
+  @Prop() readonly?: boolean;
+  @Prop() formmethod: string;
+  @Prop() formnovalidate? = false;
+  @Prop() form: string;
+  @Prop() formenctype: string;
+  @Prop() formaction: string;
+  @Prop() formtarget: string;
+  @Event() changed: EventEmitter<number>;
+
+  /**
+   * Auto-validate and display error message on form submit
+   */
+  @Prop() autoValidate?: boolean = true;
+  /**
+   * Unique input control id
+   */
+  private inputId = randomString();
+
+  /**
+   * Custom error message if control is invalid
+   */
+  @Prop() message: string;
+  @State() isInError = false;
+  private inputElement: HTMLInputElement;
+
+  handleChange = e => {
+    console.log(e.target.value);
+    this.checked = true;
+
+    // this.toggle.emit(this.checked);
+  };
+  @Method()
+  async validate(): Promise<boolean> {
+    this.isInError = !(await this.checkValidity());
+
+    return this.isInError;
+  }
+
+  /**
+   * Check if the control is valid
+   */
+  @Method()
+  async checkValidity(): Promise<boolean> {
+    return this.inputElement.checkValidity();
+  }
+
+  render() {
+    // Shouldn't overwrite immutable properties so we must use local variables
+
+    const variant = this.variant;
+    const size = this.size;
+    //let imagePosition = this.imagePosition;
+    if (!['standard', 'small'].includes(size)) {
+      throw Error('Check size must be either standard or small.');
+    }
+    if (!['radios', 'buttons'].includes(variant)) {
+      console.log('radio variant must be either radio or buttons.');
+      this.variant = 'radios';
+    }
+    if (!['left', 'right'].includes(this.imagePosition)) {
+      console.log('radio group image position must be either left or right.');
+      this.imagePosition = 'left';
+    }
+
+    {
+      let image: VNode;
+      if (this.icon != null && this.checked !== true) {
+        console.log('icon');
+        image = getSvg(this.icon, { class: 'dc-radio-image' });
+      } else if (this.src != null) {
+        image = <img src={this.src} class="dc-radio-image"></img>;
+        return;
+      }
+
+      const classes = {
+        'dc-radio-disabled': this.disabled,
+        'dc-radio-checked': this.checked,
+        [`dc-radio-${variant}`]: true,
+        [`dc-radio-size-${size}`]: true,
+        [this.type]: true,
+      };
+      if (this.variant == 'buttons') {
```

Comment:
Slightly better practice is to create a render function for each branch of the "if" that returns a fragment (<Fragment>). E.g. renderButtons and renderRadios. So the logic for the image button applies to buttons not radio. This would also require the classes variable to be rendered on the wrapping div or generated within each function (your choice).

This makes maintenance much easier. But this code is workable, so only do this if you want to explore this pattern.

Reply:
Thanks. I want to explore it, but I need little bit more assistance.